core/base/index_set.cpp

```diff
@@ -131,9 +149,10 @@ Array<IndexType> IndexSet<IndexType>::map_local_to_global(
 
     GKO_ASSERT(this->get_num_subsets() >= 1);
     exec->run(index_set::make_local_to_global(
-        this->index_space_size_, &this->subsets_begin_, &this->subsets_end_,
-        &this->superset_cumulative_indices_, &local_indices, &global_indices,
-        is_sorted));
+        this->get_num_subsets(), this->get_subsets_begin(),
+        this->get_superset_indices(),
+        static_cast<IndexType>(local_indices.get_num_elems()),
```

Comment:
perhaps it would be better to change the kernel parameter type to `size_type`? (Also for other similar kernels)

Reply:
I wanted to be a bit more explicit, so I prefer having `IndexType` as the type for the `num_indices` parameter. In general, we also wanted to move away from using `size_type` and using `IndexType` everywhere.